internal/stores/metadata.go

```diff
@@ -422,6 +419,60 @@ func (s *SQLStore) AncestorContracts(ctx context.Context, id types.FileContractI
 	return contracts, nil
 }
 
+func (s *SQLStore) ArchiveContract(ctx context.Context, id types.FileContractID, reason string) error {
+	return s.ArchiveContracts(ctx, map[types.FileContractID]string{id: reason})
+}
+
+func (s *SQLStore) ArchiveContracts(ctx context.Context, toArchive map[types.FileContractID]string) error {
+	// fetch ids
+	var ids []types.FileContractID
+	for id := range toArchive {
+		ids = append(ids, id)
+	}
+
+	// fetch contracts
+	cs, err := contracts(s.db, ids)
+	if err != nil {
+		return err
+	}
+
+	// archive them
+	if err := s.retryTransaction(func(tx *gorm.DB) error {
+		return archiveContracts(tx, cs, toArchive)
+	}); err != nil {
+		return err
+	}
+
+	// NOTE: for the time being, we do not remove known contracts when we
+	// archive a contract, this avoids a race condition when process consensus
+	// changes for recently archived contracts
+	//
+	// remove from known contracts for _, id := range ids {
+	//  delete(s.knownContracts, id)
```

Comment:
I don't think we need to keep this around. Removing a contract from the known contracts when they are archived was not correct to begin with apart from the race condition. Because we have always checked both the active and archived contracts in `ProcessConsensusChange`. The only reason to ever get rid of a known contract is when we delete them from the db for good. Knowing that we never want to interact with it again.